server/src/main/java/org/opentosca/toscana/plugins/kubernetes/docker/mapper/BaseImageMapper.java

```diff
@@ -0,0 +1,340 @@
+import org.opentosca.toscana.plugins.kubernetes.docker.mapper.model.DockerImageTag;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import static org.opentosca.toscana.plugins.kubernetes.docker.mapper.MapperConstants.ARCHITECTURE_MAP;
+import static org.opentosca.toscana.plugins.kubernetes.docker.mapper.MapperConstants.DEFAULT_IMAGE_DISTRO;
+import static org.opentosca.toscana.plugins.kubernetes.docker.mapper.MapperConstants.DEFAULT_IMAGE_PATH;
+import static org.opentosca.toscana.plugins.kubernetes.docker.mapper.MapperConstants.VERSION_DELIMITER_REGEX;
+import static org.opentosca.toscana.plugins.kubernetes.docker.mapper.MapperUtils.TAG_COMPARATOR_MINOR_VERSION;
+import static org.opentosca.toscana.plugins.kubernetes.docker.mapper.MapperUtils.anythingSet;
+import static org.opentosca.toscana.plugins.kubernetes.docker.mapper.MapperUtils.convertToVersion;
+import static org.opentosca.toscana.plugins.kubernetes.docker.mapper.MapperUtils.hasArchitecture;
+import static org.opentosca.toscana.plugins.kubernetes.docker.mapper.MapperUtils.hasDistribution;
+import static org.opentosca.toscana.plugins.kubernetes.docker.mapper.MapperUtils.hasType;
+import static org.opentosca.toscana.plugins.kubernetes.docker.mapper.MapperUtils.hasVersion;
+import static org.opentosca.toscana.plugins.kubernetes.docker.mapper.MapperUtils.isLinux;
+
+/**
+ This class allows the mapping of OSCapabilities (form a tosca model) to a docker base image.
+ When initialized with Spring this class will automatically download the latest tags from the base images
+ defined in the <code>DockerBaseImages</code> every 24 hours (by default).
+ To update this value please modify the <code>toscana.docker.base-image-mapper.update-interval</code> property (value in
+ hours)
+ */
+@Service
+@Profile("!test")
+@SuppressWarnings("ConstantConditions")
+public class BaseImageMapper {
+
+    private static final Logger logger = LoggerFactory.getLogger(BaseImageMapper.class);
+
+    @Value("${toscana.docker.base-image-mapper.update-interval}")
+    private int updateInterval;
+
+    /**
+     This field is used to check the last update timestamp
+     */
+    private long lastUpdate;
+    /**
+     The map containing the "raw" tag data collected from dockerhub
+     */
+    private Map<String, DockerImage> imageMap = new HashMap<>();
+
+    /**
+     A array conatinig the base images that this Base Image Mapper tries to map to
+     this usually is <code>DockerBaseImages.values()</code>
+     */
+    private DockerBaseImages[] baseImages;
+
+    @Autowired
+    public BaseImageMapper(DockerBaseImages[] dockerBaseImages) {
```

Comment:
This class is huge. You should split it up.
I propose you extract the logic regarding the versioning into its own class.
Furthermore, I propose to use an external library for semantic versioning parsing - it's an often solved problem and implementing it by ourselves means more brittle code to maintain.